Tests/SentryProfilerTests/SentryContinuousProfilerTests.swift

```diff
@@ -21,32 +21,125 @@ final class SentryContinuousProfilerTests: XCTestCase {
         
+        var expectedAddresses: [NSNumber] = [0x1, 0x2, 0x3]
+        try addMockSamples(mockAddresses: expectedAddresses)
+        fixture.timeoutTimerFactory.fire()
         XCTAssert(SentryContinuousProfiler.isCurrentlyProfiling())
+        try assertValidData(expectedEnvironment: expectedEnvironment, expectedAddresses: expectedAddresses)
         
-        try addMockSamples()
-        SentryContinuousProfiler.stop()
+        expectedAddresses = [0x4, 0x5, 0x6]
+        try addMockSamples(mockAddresses: expectedAddresses)
+        fixture.timeoutTimerFactory.fire()
+        XCTAssert(SentryContinuousProfiler.isCurrentlyProfiling())
+        try assertValidData(expectedEnvironment: expectedEnvironment, expectedAddresses: expectedAddresses)
         
+        expectedAddresses = [0x7, 0x8, 0x9]
+        try addMockSamples(mockAddresses: expectedAddresses)
+        XCTAssert(SentryContinuousProfiler.isCurrentlyProfiling())
+        SentryContinuousProfiler.stop()
         XCTAssertFalse(SentryContinuousProfiler.isCurrentlyProfiling())
 
-        //swiftlint:disable todo
-        // TODO: assert valid continuous profiling data when schema changes are implemented
-        //swiftlint:enable todo
+        try assertValidData(expectedEnvironment: expectedEnvironment, expectedAddresses: expectedAddresses)
     }
     
-    func addMockSamples(threadMetadata: SentryProfileTestFixture.ThreadMetadata = SentryProfileTestFixture.ThreadMetadata(id: 1, priority: 2, name: "main"), addresses: [NSNumber] = [0x3, 0x4, 0x5]) throws {
-        let state = try XCTUnwrap(SentryContinuousProfiler.profiler()?.state)
-        fixture.currentDateProvider.advanceBy(nanoseconds: 1)
-        SentryProfilerMocksSwiftCompatible.appendMockBacktrace(to: state, threadID: threadMetadata.id, threadPriority: threadMetadata.priority, threadName: threadMetadata.name, addresses: addresses)
-        fixture.currentDateProvider.advanceBy(nanoseconds: 1)
-        SentryProfilerMocksSwiftCompatible.appendMockBacktrace(to: state, threadID: threadMetadata.id, threadPriority: threadMetadata.priority, threadName: threadMetadata.name, addresses: addresses)
-        fixture.currentDateProvider.advanceBy(nanoseconds: 1)
-        SentryProfilerMocksSwiftCompatible.appendMockBacktrace(to: state, threadID: threadMetadata.id, threadPriority: threadMetadata.priority, threadName: threadMetadata.name, addresses: addresses)
+    func assertValidData(expectedEnvironment: String, expectedAddresses: [NSNumber]?) throws {
+        let envelope = try XCTUnwrap(self.fixture.client?.captureEnvelopeInvocations.last)
+        XCTAssertEqual(1, envelope.items.count)
+        let profileItem = try XCTUnwrap(envelope.items.first)
+        XCTAssertEqual("profile_chunk", profileItem.header.type)
+        let data = profileItem.data
+        let profile = try XCTUnwrap(try JSONSerialization.jsonObject(with: data) as? [String: Any])
+
+        XCTAssertNotNil(profile["version"])
```

Comment:
`l`: Asserting for the expected version depending on continuous profiling or tracer based profiling would improve the tests.